Deprecate config item `max-memory` and add items `server-memory-quota` and `memory-usage-alarm-ratio` (#4977)
pingcap/docs-cn: tidb-configuration-file.md

+ 默认值：0
+ 默认值为 0 表示使用机器上所有的 CPU；如果设置成 n，那么 TiDB 会使用 n 个 CPU 数量。

### `server-memory-quota`

> + TiDB 内存使用限制，单位为字节。

Reviewer: TiDB --> tidb-server 实例

+ 默认值：0
+ 默认值为 0 表示使用机器上所有的 CPU；如果设置成 n，那么 TiDB 会使用 n 个 CPU 数量。

> ### `max-memory`

Reviewer 2: 可能有用户还在使用这个配置项，废弃这个配置项是否需要说明一下呢？
+ 不推荐使用/已不能使用/已被新的配置项替代？
+ 从哪个版本废弃的？

> + tidb-server 实例内存使用限制，单位为字节。

Reviewer 2: ```suggestion
+ tidb-server 实例内存的使用限制，单位为字节。
```

+ 默认值：0 
+ 默认值为 0 表示无内存限制。

> > **注意：**

Reviewer 2: 放到 L282 会更醒目。
```suggestion
> **警告：**
```

>

> > `server-memory-quota` 目前为实验性特性，不建议生产环境使用。

Reviewer 3: ```suggestion
> `server-memory-quota` 目前为实验性特性，不建议在生产环境中使用。
```

> + tidb-server 实例内存的使用限制，单位为字节。(原配置项 `max-memory` 已被该配置项代替)

Reviewer 2: 原配置项 `max-memory` 被一个实验特性配置项所替代？

### `memory-usage-alarm-ratio`

+ tidb-server 实例内存使用占总内存的比例超过一定阈值时会报警。该配置项的有效范围为 0 到 1。如果配置该选项为 0 或 1，则表示关闭内存阈值报警功能。
+ 默认值：0.8
+ 当内存阈值报警功能开启时，如果配置项 [`server-memory-quota`](/tidb-configuration-file.md#server-memory-quota) 未设置，则内存报警阈值为 `memory-usage-alarm-ratio * 系统内存大小`；如果 `server-memory-quota` 被设置且大于 0，则内存报警阈值为 `memory-usage-alarm-ratio * server-memory-quota`。

> + 当 TiDB 检测到内存使用超过了阈值，则会认为存在 OOM 的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。

Reviewer 2: ```suggestion
+ 当 TiDB 检测到 tidb-server 的内存使用超过了阈值，则会认为存在内存溢出的风险，会将当前正在执行的所有 SQL 语句中内存使用最高的 10 条语句和运行时间最长的 10 条语句以及 heap profile 记录到目录 [`tmp-storage-path/record`](/tidb-configuration-file.md#tmp-storage-path) 中，并输出一条包含关键字 `tidb-server has the risk of OOM` 的日志。
```
